enterprise/server/sociartifactstore/sociartifactstore_test.go

```diff
@@ -317,3 +348,22 @@ func (r *containerRegistry) Address() string {
 func (r *containerRegistry) ImageAddress(imageName string) string {
 	return fmt.Sprintf("%s:%d/%s", r.host, r.port, imageName)
 }
+
+type deduper struct {
+	mu sync.Mutex
+}
+
+func (d *deduper) Do(_ context.Context, _ string, work func() ([]byte, error)) ([]byte, error) {
+	d.mu.Lock()
```

Comment:
Why do you need a lock here?

Reply:
It's not needed, I just added it to be slightly more consistent with the deduper contract but am happy to just remove it instead.